Sources/MapboxCoreNavigation/NavigationRouter.swift

```diff
@@ -0,0 +1,179 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+
+public class NavigationRouter {
```

Comment:
This class isn't thread-safe, please consider making it usable from any thread. Is there any reason to keep this class non-thread-safe?

As far as I see, only `activeRequests` property is modified. Protecting it with a lock would be sufficient to make this class thread safe, hopefully. NavNative implementation should be thread-safe.

Reply:
Right, I didn't bother about thread-safety so far since it's at draft stage. But since There is no feedback against the approach in general, I've added it now. NN methods use their worker threads, so it should be thread-safe by default, but I also wrapped it's calls for extra safety